web3.js/src/transaction.ts

```diff
@@ -259,9 +259,12 @@ export class Transaction {
 
     // Sort. Prioritizing first by signer, then by writable
     accountMetas.sort(function (x, y) {
+      const pubkeySorting = x.pubkey
+        .toBase58()
+        .localeCompare(y.pubkey.toBase58());
```

Comment:
can i ask why localeCompare? These are public keys, not sure what internationalization has to do with it? 🤔

Reply:
I didn't notice this in review but agreed that it's overkill here, direct comparison should be fine here. It's fine as is though, imo

Reply:
it breaks in react native depending on which JS engine u use, that's how i noticed it. Might be worth refactoring

Reply:
Oh ok, let's refactor then. Do you have the bandwidth to PR the change?

Reply:
@jstarry i'll ask someone on my team to upstream the fix 👍

Reply:
Excellent, thanks!

Reply:
@jstarry @mvayngrib apologizes for the delay, but I just created a [PR](https://github.com/solana-labs/solana/pull/28234) to upstream the fix.